src/_common/comment/comment-store.ts

```diff
@@ -86,224 +63,207 @@ export class CommentStoreModel {
+		const lastTimestamp =
+			lastComment !== null && !lastComment.is_pinned ? lastComment.posted_on : null;
+
+		response = await fetchComments(store.resource, store.resourceId, store.sort, {
+			scrollId: lastTimestamp,
+		});
+	} else {
+		// 'hot' and 'top' paginate
+		response = await fetchComments(store.resource, store.resourceId, store.sort, {
+			page: page || 1,
+		});
 	}
 
-	@VuexAction
-	async fetchComments(payload: CommentActions['comment/fetchComments']) {
-		const { store, page } = payload;
-		let response: any;
-
-		// 'new' and 'you' sort by last timestamp using scroll
-		if (store.sort === Comment.SORT_NEW || store.sort === Comment.SORT_YOU) {
-			// load comments after the last timestamp
-			const lastComment =
-				store.parentComments.length === 0
-					? null // no comments loaded
-					: store.parentComments[store.parentComments.length - 1];
-
-			// only use the last comment's timestamp if it's not pinned (pinned comment's dates are sorted differently)
-			const lastTimestamp =
-				lastComment !== null && !lastComment.is_pinned ? lastComment.posted_on : null;
-
-			response = await fetchComments(store.resource, store.resourceId, store.sort, {
-				scrollId: lastTimestamp,
-			});
-		} else {
-			// 'hot' and 'top' paginate
-			response = await fetchComments(store.resource, store.resourceId, store.sort, {
-				page: page || 1,
-			});
-		}
+	const count = response.count || 0;
+	const parentCount = response.parentCount || 0;
+	const comments = Comment.populate(response.comments).concat(
+		Comment.populate(response.childComments)
+	);
 
-		const count = response.count || 0;
-		const parentCount = response.parentCount || 0;
-		const comments = Comment.populate(response.comments).concat(
-			Comment.populate(response.childComments)
-		);
+	setCommentCount(store, count);
+	_setParentCommentCount(store, parentCount);
+	_addComments(store, comments);
 
-		this.setCommentCount({ store, count });
-		this.setParentCommentCount({ store, count: parentCount });
-		this._addComments({ store, comments });
+	return response;
+}
 
-		return response;
-	}
+export async function pinComment(manager: CommentStoreManager, comment: Comment) {
+	await comment.$pin();
 
-	@VuexAction
-	async pinComment(payload: CommentActions['comment/pinComment']) {
-		const { comment } = payload;
-		await comment.$pin();
-		const store = this.getCommentStore(comment.resource, comment.resource_id);
-		if (store instanceof CommentStoreModel) {
-			store.afterModification();
-		}
+	const store = getCommentStore(manager, comment.resource, comment.resource_id);
+	if (store instanceof CommentStoreModel) {
+		store.afterModification();
 	}
+}
 
-	@VuexAction
-	async setSort(payload: CommentActions['comment/setSort']) {
-		const { store, sort } = payload;
-		store.sort = sort;
-		// clear the store's comments and prepare for reload
-		store.clear();
-	}
+export function setSort(store: CommentStoreModel, sort: string) {
```

Comment:
Let's call this setCommentSort or something. Let' stry to get "comment" in the name of all these exported functions to make imports easier.